api-registration/server/index.js

```diff
@@ -0,0 +1,57 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+  argon2
+    .hash(password)
+    .then(hashedPassword => {
+      handleHashed(hashedPassword);
+    })
+    .catch(err => console.error(err));
```

Comment:
We alway want to make sure the client receives a response. If there are errors here, they'll be printed, but the client request will hang and the server will leak socket connections. Pass your `err` to `next()` so that the error handling middleware can deal with it and send a response.